src/Home/useBluetoothStatus.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react"
+import { Platform, AppState } from "react-native"
+import { isBluetoothEnabled } from "../gaen/nativeModule"
+
+export const useBluetoothStatus = (): boolean => {
+  const [btStatus, setBTStatus] = useState(false)
+  useEffect(() => {
+    const fetchBTEnabled = async () => {
+      const status = await fetchBTStatus()
+      setBTStatus(status)
+    }
+
+    AppState.addEventListener(getOSListener(), () => fetchBTEnabled())
```

Comment:
Hooking into AppState because certain change events, such as toggling BT via the Android notification bar, do not affect the React component lifecycle and will not cause a rerender/update.